pydatastructs/graphs/algorithms.py

```diff
@@ -637,3 +639,65 @@ def _depth_first_search_adjacency_list(
                 return None
 
 _depth_first_search_adjacency_matrix = _depth_first_search_adjacency_list
+
+def random_walk(graph, start, path_length, operation, seed=44, *args):
+    '''
+    Executes random walk from the starting node and given graph.
+
+    Parameters
+    ==========
+
+    graph: Graph
+        The graph on which random walk needs to be executed.
+    start: GraphNode or it's subclasses
+        The node/vertex from where random walk will begin.
+    path_length:
+        The maximum number of nodes/vertices traversed by random walk.
+    operation: function
+        The function which is to be applied
+        on next node, in order to append it to path.
+        The prototype which is to be followed is,
+        `function_name(next_node,
+                       arg_1, arg_2, . . ., arg_n)`.
```

Comment:
Doesn't look good.

Reply:
Should I explain it more? Or improve the language?